v3_async_wip/v3_async_wip/iothub_mqtt_client.py

```diff
@@ -275,6 +239,15 @@ async def disconnect(self) -> None:
         await self._mqtt_client.disconnect()
         logger.debug("Disconnect succeeded")
 
+    async def report_connection_drop(self) -> MQTTError:
```

Comment:
>report_connection_drop [](http://example.com/codeflow?start=14&length=22)

Confusing name. This function does not report a connection drop. It waits for a connection drop.

Reply:
I wanted to add something about the fact it returns the error. Let me think about some other names

Reply:
maybe `wait_for_connection_drop` since the waiting is the important part.